cmd/crossplane/main.go

```diff
@@ -19,205 +19,47 @@ package main
 import (
 	"os"
 	"path/filepath"
-	"time"
 
-	"github.com/go-logr/logr"
-	"github.com/spf13/afero"
 	"gopkg.in/alecthomas/kingpin.v2"
-	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
-	"k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
 	"github.com/crossplane/crossplane-runtime/pkg/logging"
-	"github.com/crossplane/crossplane/apis"
-	"github.com/crossplane/crossplane/pkg/controller/oam"
-	"github.com/crossplane/crossplane/pkg/controller/stacks"
-	"github.com/crossplane/crossplane/pkg/controller/stacks/templates"
-	"github.com/crossplane/crossplane/pkg/controller/workload"
-	stack "github.com/crossplane/crossplane/pkg/stacks"
-	"github.com/crossplane/crossplane/pkg/stacks/walker"
+	"github.com/crossplane/crossplane/cmd/crossplane/core"
+	"github.com/crossplane/crossplane/cmd/crossplane/stack/manage"
+	"github.com/crossplane/crossplane/cmd/crossplane/stack/unpack"
 )
 
-type fullCommand struct {
-	clause  *kingpin.CmdClause
-	handler func()
-}
-
-type appReq struct {
-	app        *kingpin.Application
-	zl         logr.Logger
-	syncPeriod *time.Duration
-}
-
 func main() {
-	baseCmd := filepath.Base(os.Args[0])
-	args := os.Args[1:]
-
-	app := kingpin.New(baseCmd, "An open source multicloud control plane.").DefaultEnvars()
-	debug := app.Flag("debug", "Run with debug logging.").Short('d').Bool()
-	syncPeriod := app.Flag("sync", "Controller manager sync period duration such as 300ms, 1.5h or 2h45m").Short('s').Default("1h").Duration()
-
-	a := appReq{
-		app:        app,
-		zl:         zap.New(zap.UseDevMode(*debug)),
-		syncPeriod: syncPeriod,
-	}
-
-	commands := []fullCommand{}
-	commands = append(commands, a.crossplaneCmd(baseCmd)...)
-	commands = append(commands, a.stackCmd()...)
-
-	cmdGiven := kingpin.MustParse(app.Parse(args))
-
+	var (
+		app   = kingpin.New(filepath.Base(os.Args[0]), "An open source multicloud control plane.").DefaultEnvars()
+		debug = app.Flag("debug", "Run with debug logging.").Short('d').Bool()
+		stack = app.Command("stack", "Perform operations on stacks")
+	)
+
+	c := core.FromKingpin(app.Command("core", "Start core Crossplane controllers.").Default())
```

Comment:
The new/default `core` subcommand is worth pointing out.

Reply:
This subcommand was effectively there already, wasn't it? It just happened to derive its name from the binary name previously.

Reply:
I stand corrected.

```
crossplane/crossplane (master) $ go run cmd/crossplane/main.go main --help
usage: main main
```